### PR TITLE
Adding CMake integration to EVerest

### PR DESCRIPTION
**Pull request by SebaLukas**
- Now openv2g no longer needs to be installed globally.
- Changed the folder structure.
- The header files are now in a separate include folder.
- Cleaned up the folders and files that are no longer needed

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,25 +2,40 @@
...
+    PATHS ../everest-cmake
+)
 
-include_directories(src/appHandshake src/codec src/din src/iso1 src/iso2 src/transport src/xmldsig)
+# options
+option(OPENV2G_INSTALL "Install the library (shared data might be installed anyway)" ${EVC_MAIN_PROJECT})
 
-file(GLOB SRC_FILES src/appHandshake/* src/codec/* src/din/* src/iso1/* src/iso2/* src/transport/* src/xmldsig/*)
-add_library(openv2g SHARED ${SRC_FILES})
-set_target_properties(openv2g PROPERTIES SOVERSION ${SOVERSION})
```
Comment (hikinggrass) [inline CMakeLists.txt:20]: Is the a reason SOVERSION has been removed here?

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,25 +2,40 @@
...
 
-cmake_minimum_required (VERSION 2.8)
-project (openv2g C)
+cmake_minimum_required(VERSION 3.14)
 
-set(OPENV2G_MAJOR_VERSION 0)
-set(OPENV2G_MINOR_VERSION 9)
-set(OPENV2G_PATCH_VERSION 5)
-set(VERSION ${OPENV2G_MAJOR_VERSION}.${OPENV2G_MINOR_VERSION}.${OPENV2G_PATCH_VERSION})
-set(SOVERSION 1)
```
Comment (hikinggrass) [inline CMakeLists.txt:12]: SOVERSION is used in lib/CMakeLists.txt i don't think this should be removed

File: CMakeLists.txt [hunk 3]
```diff
@@ -2,25 +2,40 @@
...
+        DIRECTORY include/
+        TYPE INCLUDE
+    )
+
+    evc_setup_package(
+        NAME everest-openv2g
+        NAMESPACE everest
+        EXPORT openv2g-targets
+    )
+endif()
```
Comment (hikinggrass) [inline CMakeLists.txt]: newline missing at end of file

File: lib/appHandshake/appHandEXIDatatypesDecoder.c [hunk 4]
```diff
@@ -858,13 +859,13 @@ static int decode_appHandAnonType_supportedAppProtocolRes(bitstream_t* stream, s
...
-
-
-int decode_appHandExiDocument(bitstream_t* stream, struct appHandEXIDocument* exiDoc) {
-	errn = readEXIHeader(stream);
-
-	if(errn == 0) {
+
+
+
+int decode_appHandExiDocument(bitstream_t* stream, struct appHandEXIDocument* exiDoc) {
```
Comment (hikinggrass) [inline lib/appHandshake/appHandEXIDatatypesDecoder.c:865]: There a quite a lot of these format changes in almost any of the modified files, this adds a lot of noise to this PR

**Review by hikinggrass — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (SebaLukas) [reply CMakeLists.txt:20]: set_target_properties works only in the lib/CMakeLists.txt

File: CMakeLists.txt (hunk 2 above)
Comment (SebaLukas) [reply CMakeLists.txt:12]: Correct, i changed that back

File: lib/appHandshake/appHandEXIDatatypesDecoder.c (hunk 4 above)
Comment (SebaLukas) [reply lib/appHandshake/appHandEXIDatatypesDecoder.c:865]: CRLF to LF causes these problems...

**Review by hikinggrass — APPROVED** [review]

**Pull request merged**